installscripts/cookbooks/jenkins/attributes/default.rb

```diff
@@ -1,14 +1,40 @@
 default['jenkins']['home'] = '/var/lib/jenkins'
 default['chef_root'] = '/tmp/jazz-chef'
-default['cookbook_root'] = "#{node['chef_root']}/cookbooks"
+default['script_root'] = "#{node['chef_root']}/chefscripts"
 default['jenkins']['clientjar'] = "#{node['chef_root']}/jenkins-cli.jar"
-default['authfile'] = "#{node['cookbook_root']}/jenkins/files/default/authfile"
-default['jenkins']['scriptApprovalfile'] = "#{node['cookbook_root']}/jenkins/files/scriptapproval/scriptApproval.xml"
-default['jenkins']['scriptApprovalfiletarget'] = "#{node['jenkins']['home']}/scriptApproval.xml"
+default['authfile'] = "#{node['chef_root']}/authfile"
+default['jenkinselb'] = 'jazz-jenkinselb-.us-east-1.elb.amazonaws.com'
+
+#This is the universal invocation that all scripts will use/append to.
+default['jenkins']['clicommand'] = "'java -jar #{node['jenkins']['clientjar']} -s http://#{node['jenkinselb']}/ -auth @#{node['authfile']}'"
+
 default['scm'] = 'bitbucket'
 default['scmelb'] = 'jazz-bitbucketelb.us-east-1.elb.amazonaws.com'
 default['scmpath'] = "#{node['scmelb']}/scm"
-default['jenkinselb'] = 'jazz-jenkinselb-.us-east-1.elb.amazonaws.com'
 default['region'] = 'us-east-1'
 default['git_branch'] = 'master'
 default['git_repo'] = 'https://github.com/tmobile/jazz.git'
+default['aws_access_key'] = 'REPLACEME'
+default['aws_secret_key'] = 'REPLACEME'
+default['cognitouser'] = 'REPLACEME'
+default['cognitopassword'] = 'REPLACEME'
+default['gitlabtoken'] = 'REPLACEME'
+default['gitlabuser'] = 'REPLACEME'
+default['gitlabpassword'] = 'REPLACEME'
+default['bbuser'] = 'REPLACEME'
+default['bbpassword'] = 'REPLACEME'
+
+#Maven cookbook property
+default['maven']['version'] = '3.5.2'
+default['maven']['setup_bin'] = true
+
+#Node cookbook property
+default['nodejs']['version'] = '8'
+default['nodejs']['install_method'] = 'package'
+#This monkeypatch is necessary because the node cookbook brokenly defaults to a 6.x sources.list otherwise
+case node['platform_family']
+when 'debian'
+  override['nodejs']['repo'] = 'https://deb.nodesource.com/node_8.x'
+when 'rhel', 'amazon'
+  default['nodejs']['repo'] = "https://rpm.nodesource.com/pub_8.x/el/#{release_ver}/$basearch"
```

Comment:
> 0m^[[0mnull_resource.chef_provision_jenkins_server (remote-exec): Chef Client failed. 0 resources updated in 03 seconds^[[0m
^[[0m^[[0mnull_resource.chef_provision_jenkins_server (remote-exec): [2018-06-22T10:08:03+00:00] FATAL: Stacktrace dumped to /tmp/jazz-chef/chef-export/.chef/local-mode-cache/cache/chef-stacktrace.out
^[[0m^[[0mnull_resource.chef_provision_jenkins_server (remote-exec): [2018-06-22T10:08:03+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
^[[0m^[[0mnull_resource.chef_provision_jenkins_server (remote-exec): [2018-06-22T10:08:03+00:00] FATAL: NoMethodError: undefined method `release_ver' for #<Chef::Node::Attribute:0x0000000004dc8530>